docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,596 @@
+
+This will create a `.devcontainer` folder in your workspace, with a `devcontainer.json` file.
+
+5. Select the "Reopen in Container" button from the notification that will show up on the bottom right corner once the dev container configuration is added to your workspace, under the `.devcontainer` folder.
+
+    You can learn more about dev containers configuration in the [documentation](https://code.visualstudio.com/docs/devcontainers/containers#_create-a-devcontainerjson-file).
+
+Once the container is set up, you will notice an indicator on the bottom left corner of VS Code:
+![Dev Container indicator displayed on the bottom left corner of VS Code.](images/fastapi-tutorial/devcontainer_indicator.png)
+
+You should now be ready to move on to the next section, where we will replace our data storage.
+
+## Replace the database
+1. In the `main.py` file, replace the `grocery_list = {}` in the beginning of the file with the lines below:
+    ```python
+    redis_client = redis.StrictRedis(host='0.0.0.0', port=6379, db=0, decode_responses=True)
+    ```
+Pylance will display an error message because redis hasn’t been imported yet.
+
+2. Click on "redis" in the editor, and select the light bulb the shows up (or `kb(editor.action.quickFix)`). Then select "Add 'import redis'".
+    ![Light bulb displayed next to the redis variable, with the option to add the import statement.](images/fastapi-tutorial/fastapi_add_redis_quickfix.png)
+
+You can also set up Pylance to automatically add imports by looking for the "Auto Import Completions" setting in the Settings UI page (`kb(workbench.action.openSettings)`) and enabling it.
+
+We now have a Redis client object that connects to a Redis server running on the local host (`host="0.0.0.0"`) and listening on port 6379 (`port=6379`). The `db` parameter specifies the Redis database to use. Redis supports multiple databases, and in this code we’re setting to use database 0, which is the default database. We're also passing `decode_responses=True` for the responses to be decoded as strings (instead of bytes).
+
+This client can now be used to perform operations such as getting, setting, and deleting objects in the Redis database.
+
+Let's do some more replacements in our first route `add_item`. Instead of looking into all the keys from our dictionary to find the item name that has been provided, we can fetch that information directly from a redis hash. Let’s assume we will create a hash called `item_name_to_id`, with item names mapped into their IDs once they are added in our app. We can then get the id of the item name we’re receiving by invoking the `hget` method from redis.
+
+3. Delete line 17 (TODO: verify line number):
+    ```python
+    items_names = list(grocery_list.values())
+    ```
+    And replace it with:
+    ```python
+      item_id = redis_client.hget("item_name_to_id", item.item_name)
```

Comment:
The about of red squiggles that showed up after this change made me uncomfortable. I wonder if there's a more incremental approach we could take to these changes?